File: UVa/Q299 Train Swapping.cpp
#include<bits/stdc++.h>
using namespace std;
int n,l,ans,arr[55];
int main(){
	cin>>n;
	while(n--){
		ans=0;
		cin>>l;
		for(int i=0;i<l;i++){
			cin>>arr[i];
		}
		for(int i=0;i<l-1;i++){
			for(int j=0;j+1<l;j++){
				if(arr[j]>arr[j+1]){
					swap(arr[j],arr[j+1]);
					ans++;
				}
			}
		}
		cout<<"Optimal train swapping takes "<<ans<<" swaps.\n";
	}
	return 0;
}

File: UVa/Q299 Train Swapping.py
t = int(input())
while t:
	t -= 1
	ans = 0
	n = int(input())
	train = [int(x) for x in input().split()]
	for i in range(n):
		for j in range(n - i - 1):
			if train[j] > train[j + 1]:
				ans += 1
				train[j],train[j + 1] = train[j + 1],train[j]
	print(f'Optimal train swapping takes {ans} swaps.')